purchase_work_acceptance/models/account_invoice.py

```diff
@@ -0,0 +1,169 @@
+                if self.reference else vendor_ref
+
+        if not self.invoice_line_ids:
+            self.currency_id = self.wa_id.currency_id
+
+        new_lines = self.env['account.invoice.line']
+        for line in self.wa_id.wa_line_ids:
+            data = self._prepare_invoice_line_from_wa_line(line)
+            new_line = new_lines.new(data)
+            new_line._set_additional_fields(self)
+            new_lines += new_line
+
+        self.invoice_line_ids += new_lines
+        self.payment_term_id = self.wa_id.purchase_id.payment_term_id
+        self.purchase_id = self.wa_id.purchase_id.id
+        self.origin = self.wa_id.purchase_id.name
+        self.env.context = dict(self.env.context, from_wa_id_change=True)
+        return {}
+
+    @api.onchange('currency_id')
+    def _onchange_currency_id(self):
+        if self.currency_id:
+            for line in self.invoice_line_ids.filtered(lambda r: r.wa_line_id):
+                date = self.date or self.date_invoice or fields.Date.today()
+                company = self.company_id
+                line.price_unit = line.wa_id.currency_id._convert(
+                    line.wa_line_id.price_unit, self.currency_id, company,
+                    date, round=False)
+
+    @api.model
+    def create(self, vals):
+        invoice = super(AccountInvoice, self).create(vals)
+        wa = invoice.invoice_line_ids.mapped('wa_line_id.wa_id')
+        if wa and not invoice.refund_invoice_id:
+            message = _(
+                "This vendor bill has been created from: %s"
+                ) % (",".join(
+                    ["<a href=# data-oe-model=work.acceptance data-oe-id=" +
+                     str(line.id) + ">" + line.name + "</a>" for line in wa]))
+            invoice.message_post(body=message)
+        return invoice
+
+    @api.multi
+    def write(self, vals):
+        result = True
+        for invoice in self:
+            wa_old = invoice.invoice_line_ids.mapped('wa_line_id.wa_id')
+            result = result and super(AccountInvoice, invoice).write(vals)
+            wa_new = invoice.invoice_line_ids.mapped('wa_line_id.wa_id')
+            wa = (wa_old | wa_new) - (wa_old & wa_new)
+            if wa:
+                message = _(
+                    "This vendor bill has been modified from: %s"
+                    ) % (",".join(
+                        ["<a href=# data-oe-model=work.acceptance"
+                         "data-oe-id=" + str(wa.id) + ">" + wa.name + "</a>"
+                         for line in wa]))
+                invoice.message_post(body=message)
+        return result
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    wa_line_id = fields.Many2one(
+        comodel_name='work.acceptance.line',
+        string='Work Acceptance Line',
+        ondelete='set null',
+        index=True,
+        readonly=True
+    )
+    wa_id = fields.Many2one(
```

Comment:
Do we need to create this field? 
Note: I see that there are many fields that also not needed (store=True fields) also in other model. Make sure that it is really necessary before we have it, then have it.